### PR TITLE
feat: configurable sibling matchers

### PR DESCRIPTION
**Pull request by olensmar**
This PR adds the possibility to configure siblingMatchers (#1605) and fixes the corresponding cert-manager issue (#1606)

image

- [X] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [X] added a test

### CODE REVIEW

File: src/models/resourcekindhandler.ts [hunk 1]
```diff
@@ -19,10 +19,14 @@ interface RefMapper {
...
         targetResource: K8sResource,
         value: string,
-        siblingValues: Record<string, string>
+        siblingValues: Record<string, string>,
+        properties?: any
       ) => boolean
     >;
 
+    // optional matcher configuration properties that will be passed to each correspondingly named matcher
+    matcherProperties?: Record<string, any>;
```
Comment (devcatalin) [inline src/models/resourcekindhandler.ts:28]: Maybe we should refactor this to:
```ts
type Matcher = 'kindMatcher' | 'groupMatcher' // ... add additional matcher types

import {JsonObject} from "type-fest"; // using this type to ensure RefMappers will remain serializable

interface RefMapper {
  // ...
  matcherProperties: Record<Matcher, JsonObject>
  // ...
}

**Review by devcatalin — CHANGES_REQUESTED** [review]

File: src/models/resourcekindhandler.ts (hunk 1 above)
Comment (olensmar) [reply src/models/resourcekindhandler.ts:28]: sure - I think this would be fine until we allow plugins to provide their own matchers... !?

Comment (olensmar) [reply src/models/resourcekindhandler.ts:28]: actually - the key of this map is dynamic - i.e. depends on the property to be matched - so we can't create a type with a fixed set of names. I'll add the JsonObject though 👍

**Review by devcatalin — APPROVED** [review]
LGTM

**Pull request merged**